File: java/acronym/src/main/java/Acronym.java
class Acronym {

    String phrase;

    Acronym(String phrase) {
        this.phrase = phrase;
    }

    String get() {
        StringBuilder acronym = new StringBuilder();

        for(String word : phrase.split(" |-")){
            acronym.append(word.toUpperCase().charAt(0));
        }

        return acronym.toString();
    }

}

File: csharp/acronym/Acronym.cs
using System;
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase) => new string(
        phrase.Split(new char[] { ' ', '-' }, StringSplitOptions.RemoveEmptyEntries)
            .Select(FirstLetter)
            .Select(char.ToUpper)
            .ToArray()
    );

    private static char FirstLetter(string str)
        => str.First(char.IsLetter);
}